### PR TITLE
Adding DFT data classes 

### PR DESCRIPTION
**Pull request by advikavs**
# Pull Request Template

## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/data/dft_data.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+
+    References
+    ----------
+    Kasim, Muhammad F., and Sam M. Vinko. "Learning the exchange-correlation functional from nature with fully differentiable density functional theory." Physical Review Letters 127.12 (2021): 126403.
+
+    https://github.com/diffqc/dqc/blob/0fe821fc92cb3457fb14f6dff0c223641c514ddb/dqc/system/base_system.py
+    """
+
+    def __init__(self, system: Dict):
+        self.system = system
```
Comment (rbharath) [inline deepchem/data/dft_data.py:61]: You should remove the system dict and make it explicit arguments as well

File: deepchem/feat/dft_data.py [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    -----
+    Entry class should not be initialized directly, but created through
+    ``Entry.create``
+
+    Example
+    -------
+    >>> from deepchem.feat.dft_data import DFTEntry
+    >>> e_type= 'dm'
+    >>> true_val= 'deepchem/data/tests/dftHF_output.npy'
+    >>> systems = [{
```
Comment (rbharath) [inline deepchem/feat/dft_data.py:88]: Doctest is complaining here. I think you want to move these all onto one line. Try running doctest locally to check correctness

**Review by rbharath — COMMENTED** [review]
Almost there, but a couple of minor points on yapf, doctest still. Let's discuss offline to get this across the finish line

**Comment by rbharath** [comment]
There are some mysterious CI errors. Are we seeing similar errors on other PRs?

**Review by rbharath — COMMENTED** [review]
@advikavs Is the CI looking good? Are there any last failing tests that we need to address?

**Comment by advikavs** [comment]
> @advikavs Is the CI looking good? Are there any last failing tests that we need to address?

I fixed the doctest error. 20/21 are passing. The only one that's failing is the yapf for the .npy file.

**Comment by advikavs** [comment]
re-opened

**Pull request closed without merging**